provider/rfc2136_test.go

```diff
@@ -98,7 +93,14 @@ func (r *rfc2136Stub) IncomeTransfer(m *dns.Msg, a string) (env chan *dns.Envelo
 }
 
 func createRfc2136StubProvider(stub *rfc2136Stub) (Provider, error) {
-	return NewRfc2136Provider("", 0, "", false, "key", "secret", "hmac-sha512", true, DomainFilter{}, false, stub)
+	return NewRfc2136Provider("", 0, "", false, "key", "secret", "hmac-sha512", true, DomainFilter{}, false, stub, 300)
+}
+
+func extractAuthoritySectionFromMessage(msg fmt.Stringer) []string {
+	const searchPattern = "AUTHORITY SECTION:"
+	data := msg.String()
+	authoritySectionOffset := strings.Index(data, searchPattern)
```

Comment:
instead of data you can use directly msg.String(), WDYT?

Reply:
I extracted this part from another test and kept it as it is, however what you suggest makes more sense.